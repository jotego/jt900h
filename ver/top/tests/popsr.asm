    ; POPSR
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xsp, stack
    ld wa,0x3ff
    push wa
    pop sr
    ld bc,3
    ld wa,0x2ff
    push wa
    pop sr
    ld bc,2
    ld wa,0x100
    push wa
    pop sr
    ld bc,1

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    dw 0,0,0,0,0,0,0,0,0,0
stack:
    end