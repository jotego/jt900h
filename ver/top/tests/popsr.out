000003ff - 00000000 - 00000100 - 000002ff
00000000 - 00000001 - 00000002 - 00000003
00000000 - 00000000 - 00000000 - 00000000
00000000 - 0000babe - 00000000 - 00000000
-- index registers --
00000000 - 00000000 - 00000000 - 0000004a
SR = 0089
